### DISCUSSION TITLE
What purpose does Update have in Images?

### DISCUSSION THREAD

**Discussion by MCVenor** [stride3d/stride#1933] [Newcomers] [answered]
Hi,

im currently implementing the UI for my game.
I found an Update method in [ImageElement](https://doc.stride3d.net/4.0/en/api/Stride.UI.Controls.ImageElement.html) .

What is the intended use for this, when would i like to use it?
Any Example is welcome.

**Comment by Ethereal77** [comment]
Any `UIElement` can have time-dependent images or effects.

Currently, the `ImageElement` uses this method to update the state of the underlying `Sprite` (for animations, I imagine). Maybe it can also be used for tweening or other UI animations. And in the future, it may be possible to use custom shaders / materials for UI so it can do time-dependent effects, etc.

**Comment by MCVenor** [comment]
So its like a place that one can fill out if needed? That sounds useful, for like a shiny sprite animation when hoverd over a button.

What if i have 100 buttons and 150 icons on the screen, do i have to implement them myself in order to **avoid** them being registerd somehow and recieve update "calls" ? ( it might be closer to 40 buttons and 90 icons in my case)

**Reply by Ethereal77** [reply, marked as answer]
They are **always** "updated" by the UI system. You don't have to register or do anything at all. They are virtual methods you can override in **derived classes** if you think you need a derived specialized `ImageElement`. But unless that is your case, you can just use `ImageElement` and not override anything.

The current implementation in there is prettly lightweight, just updates the sprite if it has changed (i.e. when you set / unset the sprite via the `Source` property). [Take a look](https://github.com/stride3d/stride/blob/master-4.0/sources/engine/Stride.UI/Controls/ImageElement.cs#L99).

**Reply by MCVenor** [reply]
Great, Thank you.
